pkg/gcmanager/pod_informer.go

```diff
@@ -26,11 +56,13 @@ func (s *SpiderGC) startPodInformer() {
 		})
 		if nil != err {
 			logger.Error(err.Error())
+			innerCancel()
 			continue
 		}
-		go podInformer.Run(stopper)
+		s.informerFactory = informerFactory
+		informerFactory.Start(innerCtx.Done())
 
-		<-stopper
+		<-innerCtx.Done()
 		logger.Error("k8s pod informer broken")
 	}
 }
```

Comment:
Here are some observations and improvements that can be made to this code:

Observations:
- It seems like the method `startPodInformer` is intended to start a Kubernetes pod informer that watches and triggers handlers upon events like addition, deletion, or updates of pods in the cluster.
- The code checks if the current instance is the elected leader before starting the informer. If it's not the leader, then it waits for an interval before checking again and tries to start the informer only if it becomes the leader.
- It seems like the `informerFactory` field is a global state used by other methods in the `SpiderGC`.

Possible Improvements:
- Provide more context around the purpose of this code in comments/docs.
- When the innerCtx is cancelled due to loss of leadership, the `informerFactory` is not stopped, which might cause resource leaks. Consider using the `Stop()` method on the factory when cancelling the context.
- Use named vars instead of positional `(stopper := make(chan struct{}))`.
- The use of go-routines and channels should be assessed more carefully. For example, `innerCtx` is being passed as an argument to `Start()`, but its cancellation is done from within a closure passed into a goroutine launched within the same method. Such ownership of contexts may not be easy to track in general; consider using packages like context/co to simplify such coordination.
- The waiting duration during non-leadership could be reduced or replaced with an exponential backoff strategy. The current implementation might introduce undue latency.
- With the `informerFactory` exposed as a global state, it is difficult to reason about how it is being used elsewhere in the codebase. Consider encapsulating it within the `SpiderGC` instance or provide documentation regarding proper usage of this state.